### PR TITLE
Add support for check_suite event

### PR DESCRIPTION
**Pull request by timon**
Hi there and thanks for your action.

I'd like to use this to merge some PRs automatically once build is verified with Travis, but Travis reports build result in `check_run` and `check_suite` events. This PR adds support for `check_suite` events.

### CODE REVIEW

File: src/automerge-action.ts [hunk 1]
```diff
@@ -270,6 +271,30 @@ export class AutomergeAction {
...
+    core.debug('handleCheckSuite()')
+
+    const { action, check_suite: checkSuite } = github.context.payload
+
+    if (!action || !checkSuite) {
+      return
+    }
+
+    if (checkSuite.conclusion !== 'success') {
+      core.info(`Check suite conclusion is ${checkSuite.conclusion}, not attempting to merge.`)
```
Comment (reitermarkus) [inline src/automerge-action.ts:284]:
```suggestion
      core.info(`Conclusion for check suite ${checkSuite.id} is ${checkSuite.conclusion}, not attempting to merge.`)
```

File: src/helpers.ts [hunk 2]
```diff
@@ -142,34 +142,57 @@ export function isDoNotMergeLabel(string: string): boolean {
...
+      direction: 'desc',
+      per_page: 100,
+    })
+  ).data as PullRequest[]
+
+  return pullRequests.filter(pr => pr.head.sha === sha).map(({ number }) => number)
+}
+
+export async function pullRequestsForCheckSuite(octokit: Octokit, checkSuite: CheckSuite): Promise<number[]> {
+  let pullRequests = (checkSuite.pull_requests as PullRequest[]).map(({ number }) => number)
```
Comment (reitermarkus) [inline src/helpers.ts:170]: Why does this need to be cast here?

**Review by reitermarkus — CHANGES_REQUESTED** [review]

**Comment by reitermarkus** [comment]
Please also update the example in the ReadMe to include this.

File: src/helpers.ts (hunk 2 above)
Comment (timon) [reply src/helpers.ts:170]: Well, the code was mostly copypasted from `pullRequestsForWorkflowRun`, which has had this conversion.
I'm not good at TypeScript, so didn't change much.

Omitting the conversion leads to the error message `Object is possibly null`, so perhaps it makes sense to replace the cast with `|| []`.
I could do that if it's fine with you.

Comment (reitermarkus) [reply src/helpers.ts]: I don't think the cast is necessary anymore, try:

```ts
checkSuite.pull_requests?.map(({ number }) => number) ?? []
```

If that works, please also adjust this in `pullRequestsForWorkflowRun`.

**Comment by reitermarkus** [comment]
Thanks, @timon!

**Pull request merged**